pkg/gentrait/securityevent/model_server.go

```diff
@@ -0,0 +1,101 @@
+package securityevent
+
+import (
+	"context"
+	"strconv"
+
+	"google.golang.org/grpc"
+
+	"github.com/smart-core-os/sc-api/go/types"
+	"github.com/smart-core-os/sc-golang/pkg/resource"
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+)
+
+type ModelServer struct {
+	gen.UnimplementedSecurityEventApiServer
+	model *Model
+}
+
+func NewModelServer(model *Model) *ModelServer {
+	return &ModelServer{model: model}
+}
+
+func (m *ModelServer) Register(server *grpc.Server) {
+	gen.RegisterSecurityEventApiServer(server, m)
+}
+
+func (m *ModelServer) Unwrap() any {
+	return m.model
+}
+
+func (m *ModelServer) ListSecurityEvents(_ context.Context, req *gen.ListSecurityEventsRequest) (*gen.ListSecurityEventsResponse, error) {
+
+	// page token is just the index of where we left off (if any)
+	// this works with the current basic implementation because we only support a list of all events without filtering/sorting
+	// and the events are stored in ascending chronological order. If this either of these things change, this will need to be rethought
+	pageToken := req.GetPageToken()
+	m.model.mu.Lock()
+	defer m.model.mu.Unlock()
+	startIndex := len(m.model.allSecurityEvents)
```

Comment:
This is _really_ bad practice, mutexes have an owner and that owner is usually the type that contains them. Only the owner should directly use or reference that mutex. You need to move the functionality that needs the lock into the Model type, I guess on a new method.